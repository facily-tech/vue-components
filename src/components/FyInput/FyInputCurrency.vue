<template>
  <div>
    <fy-input
      :value="text"
      @input="update"
      v-money="money"
      v-bind.sync="$props"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';
import { VMoney } from 'v-money';

const BaseVTextField = Vue.extend({ mixins: [VTextField] });

interface options extends InstanceType<typeof BaseVTextField> {
  value: string;
  type: string;
  decimal: string;
  thousands: string;
  precision: number;
}

export default BaseVTextField.extend<options>().extend({
  name: 'fy-input-currency',

  directives: { money: VMoney },

  props: {
    value: {
      tyupe: String,
      default: '0',
    },
    type: {
      type: String,
      default: 'R$ ',
    },
    decimal: {
      type: String,
      default: ',',
    },
    thousands: {
      type: String,
      default: '.',
    },
    precision: {
      type: Number,
      default: 2,
    },
  },
  data() {
    return {
      text: this.value ? this.value : '0',
      money: {
        decimal: ',',
        thousands: '.',
        prefix: this.type,
        precision: 2,
        masked: false /* doesn't work with directive */,
      },
    };
  },
  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
