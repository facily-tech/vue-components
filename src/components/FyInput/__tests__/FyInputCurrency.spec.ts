import Vue from 'vue';
import Vuetify from 'vuetify';
import FyInputCurrency from '../FyInputCurrency.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyInputCurrency.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInputCurrency>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInputCurrency, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
