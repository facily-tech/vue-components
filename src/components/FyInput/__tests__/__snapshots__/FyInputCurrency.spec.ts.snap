// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInputCurrency.ts should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined" maxlength="24" type="R$ " decimal="," thousands="." precision="2">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">â€‹</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input data-test="input" maxlength="24" decimal="," thousands="." precision="2" id="input-3" type="R$ "></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
